src/spec/create_extension_spec.py

```diff
@@ -0,0 +1,531 @@
+                    "Should be consistent with `insertion_position_in_mm.ml` coordinate (left = ml < 0, "
+                    "right = ml > 0)."
+                ),
+                dtype="text",
+                required=False,
+            ),
+            NWBAttributeSpec(
+                name="insertion_angle_pitch_in_deg",
+                doc=(
+                    "The pitch angle of the probe at the time of insertion, in degrees. "
+                    "Pitch = rotation around left-right axis, like nodding (+ is rotating the nose upward). "
+                    "Zero is defined as the probe being parallel to an axial slice of the brain."
+                    "Yaw = rotation around dorsal-ventral axis, like shaking (+ is rotating the nose rightward). "
+                    "Roll = rotation around anterior-posterior axis, like tilting (+ is rotating the right side "
+                    "downward)."
+                ),
+                dtype="float",
+                required=False,
+            ),
+            NWBAttributeSpec(
+                name="insertion_angle_yaw_in_deg",
+                doc=(
+                    "The yaw angle of the probe at the time of insertion, in degrees. "
+                    "Yaw = rotation around dorsal-ventral axis, like shaking (+ is rotating the nose rightward). "
+                    "Zero is defined as the probe being parallel to an sagittal slice of the brain."
+                ),
+                dtype="float",
+                required=False,
+            ),
+            NWBAttributeSpec(
+                name="insertion_angle_roll_in_deg",
+                doc=(
+                    "The roll angle of the probe at the time of insertion, in degrees. "
+                    "Roll = rotation around anterior-posterior axis, like tilting (+ is rotating the right side "
+                    "downward). Zero is defined as the probe being parallel to a coronal slice of the brain. "
+                ),
+                dtype="float",
+                required=False,
+            ),
+        ],
+    )
+
+    channels_table = NWBGroupSpec(
+        neurodata_type_def="ChannelsTable",
+        neurodata_type_inc="DynamicTable",
+        doc="Metadata about the channels used in an extracellular recording from a single probe.",
+        default_name="ChannelsTable",
+        groups=[
+            NWBGroupSpec(
+                name="probe_insertion",
+                neurodata_type_inc="ProbeInsertion",
+                doc="Information about the insertion of a probe into the brain.",
+                quantity="?",
+            ),
+        ],
+        datasets=[
+            NWBDatasetSpec(
+                name="contact",
+                neurodata_type_inc="DynamicTableRegion",
+                doc="The row in a ContactsTable that represents the contact used as a channel.",
+            ),
+            NWBDatasetSpec(
```

Comment:
How does this interact with the current `reference_mode`? Are the suppoed to be mutually exclusive? As in, you can specify which channel is used as reference and/or `reference_mode` accounts for the cases when there is a global reference?

Reply:
Thanks for bringing this up. Yeah, the documentation for `reference_mode` here is unclear. 

`reference_mode` currently does two things: 
1. capture the point of reference for getting the voltage. This is commonly referred to as the ground (e.g., skull screw, grounding wire)
2. capture any special referencing done, e.g., "common average referencing" which would mean the data across all channels is averaged and subtracted from the current channel before being recorded. Some data acquisition systems can do this online and the referenced data is what is stored.

In reviewing this, I think we should have two optional fields here `ground` and `electrical_reference_description` for each of the above. I'll make that change.

If a user supplies a `reference_channel` for each channel, then `electrical_reference_description` is not needed, but could be useful as a freeform description to explain why the references were set the way they were set.

This schema does not cover the rare case where channels 1-16 have channel 17 as reference and channels 18-32 have a common average reference using only those channels, because we cannot specify None as a reference channel for only some channels...

https://plexon.com/blog-post/point-of-reference/
https://open-ephys.github.io/gui-docs/User-Manual/Plugins/Neuropixels-PXI.html#reference-selection
https://github.com/cortex-lab/neuropixels/wiki/Referencing-and-Grounding
https://www.youtube.com/watch?v=wKx_B5PR3YU&t=489s

Reply:
Yeah, just checked the new version. Looks great. Also, thanks for those links.

Out of curiosity:
Concerning the limitation that you mentioned, is that a limitation because the dtype of `DynamicTableRegion` is an int and we don't have nullable ints, right? 

On the other hand, given that this is supposed to represent indices, could we just use a negative number to indicate that?

Reply:
> a limitation because the dtype of DynamicTableRegion is an int and we don't have nullable ints, right?

Right

> On the other hand, given that this is supposed to represent indices, could we just use a negative number to indicate that?

Yes, we could. If we want to do that, then we should codify that in https://github.com/hdmf-dev/hdmf-common-schema/blob/main/common/table.yaml#L71 and guard for that in the HDMF API. I'm pretty sure a -1 row index would result in an error right now.

I think we do not need to do this until a user requests it. I think it is a pretty rare use case.

Reply:
OK, thanks for explaining this to me.